app/controllers/concerns/jwt_authorization_concern.rb

```diff
@@ -0,0 +1,62 @@
+# Inspects the HTTP Authorization header for a Bearer JWT token and
+# populates the @auth_token_payload instance variable with the JWT payload.
+# This does no signature verification of the JWT, expecting a gateway or
+# proxy to have done that already.
+# If the payload cannot be parsed or does not have a preferred_username
+# attribute, a 401 Unauthorized response with an empty JSON body will be
+# returned.
+# All private API endpoints should include this concern and can rely on the
+# preferred_username to represent the user.
+module JwtAuthorizationConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :check_token
+  end
+
+  private
+
+  def check_token
+    @auth_token_payload = nil
+    @current_user = nil
+
+    jwt = nil
+
+    auth_header = request.headers['Authorization']
+    if !auth_header.nil?
+      m = auth_header.match(/^Bearer (.*)$/)
+      if m && m.captures.length == 1
+        pieces = m.captures[0].split('.')
+        if pieces.length == 3
+          begin
+              jwt = JSON.parse(Base64.decode64(pieces[1]))
```

Comment:
If we're going to rely on a gateway maybe there's a header it will set so we can avoid looking like we're verifying the JWT when we're not?

Otherwise there are a bunch of libraries available to verify the JWT:

https://github.com/jwt/ruby-jwt
https://github.com/heartcombo/devise
https://auth0.com/docs/quickstart/backend/rails
https://www.digitalocean.com/community/tutorials/how-to-set-up-user-authentication-with-devise-in-a-rails-7-application

Reply:
ok, updated to use the jwt gem.  This also gives us a path for more verification.